core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java

```diff
@@ -132,44 +138,58 @@ protected SwaggerBundleConfiguration getSwaggerBundleConfiguration(StreamRegistr
         Preconditions.checkState(managedInfraManager != null, "managedInfraManager cannot be null.");
-
         ManagedKafkaProducer managedKafkaProducer = createManagedKafkaProducer(configuration);
 
-        // Step 2 - initialize the DAO's
+        // Step 2 - Create the Kafka topics used as DataStore
+        createKafkaTopicUsedAsDatastore(configuration);
 
+        // Step 3 - initialize the DAO's
         initServiceAndDao(configuration, environment, managedKStreams, managedKafkaProducer);
 
-        // Step 3 - initialize and register the SR resources to JerseyEnvironment
-
+        // Step 4 - initialize and register the SR resources to JerseyEnvironment
         initAndRegisterResource(environment);
 
-        // Step 4 - initialize and register monitoring and health check hooks
-
+        // Step 5 - initialize and register monitoring and health check hooks
         environment.getApplicationContext().addServlet(PingServlet.class, "/ping");
         initAndRegisterHealthCheck(configuration, environment, managedKStreams);
 
-        // Step 5 - centralize initialization and configuration of SR server Object Mapper's
+        // Step 6 - centralize initialization and configuration of SR server Object Mapper's
         registerServiceMapper(environment);
 
-        // Step 6 - In order to avoid muddle up with HK2 and Jersey lifecycle dependency
+        // Step 7 - In order to avoid muddle up with HK2 and Jersey lifecycle dependency
         // we wrap up all managed beans in a centralized container to manage server
         // components start and stop ordering
-
         registerManagedContainer(environment, managedKStreams, managedInfraManager, managedKafkaProducer);
 
-        // Step 7 - register list of Jersey Filters.
+        // Step 8 - register list of Jersey Filters.
         registerFilters(environment, configuration);
 
         // TODO: Make project completely based on unit tests (integration should be a separate project) (#100)
     }
 
+    private void createKafkaTopicUsedAsDatastore(StreamRegistryConfiguration configuration) {
+        String kafkaBootstrapURI = configuration.getKafkaProducerConfig().getKafkaProducerProperties().get("bootstrap.servers");
+        Properties properties = new Properties();
+        properties.put(BOOTSTRAP_SERVERS_CONFIG, kafkaBootstrapURI);
+        AdminClient adminClient = KafkaAdminClient.create(properties);
+
+        String topicName = configuration.getTopicsConfig().getProducerTopic();
+        try {
+            if (isKafkaTopicAlreadyPresent(adminClient, topicName)) {
+                makeSureTopicIsCompacted(adminClient, topicName);
+            } else {
+                createTopicWithCompactionEnabled(adminClient, topicName);
+            }
+        } catch (Exception e) {
+            log.error("Error while communication with the underlying data-store. Topic={} BootstrapURI={}", topicName, kafkaBootstrapURI);
+            throw new RuntimeException(e);
```

Comment:
`throw new IllegalStateException("Error while communication with the underlying data-store. Topic="+ topicName + " BootstrapURI=" + kafkaBootstrapURI, runtimeException);`